swagger2 2.9 doesn't compile with insert-ordered-containers < 0.3

It fails with

```
[ 1 of 18] Compiling Data.HashMap.Strict.InsOrd.Compat ( src/Data/HashMap/Strict/InsOrd/Compat.hs, dist/build/Data/HashMap/Strict/InsOrd/Compat.o, dist/build/Data/HashMap/Strict/InsOrd/Compat.dyn_o )
src/Data/HashMap/Strict/InsOrd/Compat.hs:80:3: error: [GHC-87543]
    Ambiguous occurrence ‘null’.
    It could refer to
       either ‘Prelude.null’,
              imported from ‘Prelude’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:74:8-40
              (and originally defined in ‘ghc-internal-9.1003.0:GHC.Internal.Data.Foldable’),
           or ‘Data.HashMap.Strict.InsOrd.null’,
              imported from ‘Data.HashMap.Strict.InsOrd’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:140:1-33.
   |
80 |   null,
   |   ^^^^

src/Data/HashMap/Strict/InsOrd/Compat.hs:83:3: error: [GHC-87543]
    Ambiguous occurrence ‘lookup’.
    It could refer to
       either ‘Prelude.lookup’,
              imported from ‘Prelude’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:74:8-40
              (and originally defined in ‘ghc-internal-9.1003.0:GHC.Internal.List’),
           or ‘Data.HashMap.Strict.InsOrd.lookup’,
              imported from ‘Data.HashMap.Strict.InsOrd’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:140:1-33.
   |
83 |   lookup,
   |   ^^^^^^

src/Data/HashMap/Strict/InsOrd/Compat.hs:97:3: error: [GHC-87543]
    Ambiguous occurrence ‘map’.
    It could refer to
       either ‘Prelude.map’,
              imported from ‘Prelude’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:74:8-40
              (and originally defined in ‘ghc-internal-9.1003.0:GHC.Internal.Base’),
           or ‘Data.HashMap.Strict.InsOrd.map’,
              imported from ‘Data.HashMap.Strict.InsOrd’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:140:1-33.
   |
97 |   map,
   |   ^^^

src/Data/HashMap/Strict/InsOrd/Compat.hs:111:3: error: [GHC-87543]
    Ambiguous occurrence ‘foldl'’.
    It could refer to
       either ‘Prelude.foldl'’,
              imported from ‘Prelude’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:74:8-40
              (and originally defined in ‘ghc-internal-9.1003.0:GHC.Internal.Data.Foldable’),
           or ‘Data.HashMap.Strict.InsOrd.foldl'’,
              imported from ‘Data.HashMap.Strict.InsOrd’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:140:1-33.
    |
111 |   foldl',
    |   ^^^^^^

src/Data/HashMap/Strict/InsOrd/Compat.hs:113:3: error: [GHC-87543]
    Ambiguous occurrence ‘foldr’.
    It could refer to
       either ‘Prelude.foldr’,
              imported from ‘Prelude’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:74:8-40
              (and originally defined in ‘ghc-internal-9.1003.0:GHC.Internal.Data.Foldable’),
           or ‘Data.HashMap.Strict.InsOrd.foldr’,
              imported from ‘Data.HashMap.Strict.InsOrd’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:140:1-33.
    |
113 |   foldr,
    |   ^^^^^

src/Data/HashMap/Strict/InsOrd/Compat.hs:120:3: error: [GHC-87543]
    Ambiguous occurrence ‘filter’.
    It could refer to
       either ‘Prelude.filter’,
              imported from ‘Prelude’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:74:8-40
              (and originally defined in ‘ghc-internal-9.1003.0:GHC.Internal.List’),
           or ‘Data.HashMap.Strict.InsOrd.filter’,
              imported from ‘Data.HashMap.Strict.InsOrd’ at src/Data/HashMap/Strict/InsOrd/Compat.hs:140:1-33.
    |
120 |   filter,
    |   ^^^^^^
```

because the [relevant branch](https://github.com/GetShopTV/swagger2/blob/614cc220745942f7adfc9d0e7b2ff0504494494a/src/Data/HashMap/Strict/InsOrd/Compat.hs#L139) doesn't hide Prelude imports.

--- a/src/Data/HashMap/Strict/InsOrd/Compat.hs
+++ b/src/Data/HashMap/Strict/InsOrd/Compat.hs
@@ -138,6 +138,7 @@ module Data.HashMap.Strict.InsOrd.Compat (
 
 #if !MIN_VERSION_insert_ordered_containers(0,3,0)
 import Data.HashMap.Strict.InsOrd
+import Prelude hiding (map, null, filter, foldr, foldl', lookup)
 #else
 import qualified Data.HashMap.Strict.InsOrd as InsOrdHashMap
 

--- a/swagger2.cabal
+++ b/swagger2.cabal
@@ -34,7 +34,7 @@ tested-with:
 
 custom-setup
   setup-depends:
-                base          >=4.9   &&< 4.22
+                base          >=4.9   &&< 4.23
               , Cabal < 3.15
               , cabal-doctest >=1.0.6 && <1.1
 
@@ -64,11 +64,11 @@ library
 
   -- GHC boot libraries
   build-depends:
-      base             >=4.9       && <4.22
+      base             >=4.9       && <4.23
     , bytestring       >=0.10.8.1  && <0.13
     , containers       >=0.5.7.1   && <0.9
-    , template-haskell >=2.11.1.0  && <2.24
-    , time             >=1.6.0.1   && <1.15
+    , template-haskell >=2.11.1.0  && <2.25
+    , time             >=1.6.0.1   && <1.17
     , transformers     >=0.5.2.0   && <0.7
 
   build-depends:
@@ -78,7 +78,7 @@ library
   -- other dependencies
   build-depends:
       base-compat-batteries     >=0.11.1   && <0.16
-    , aeson                     >=2.0.0.0  && <2.3
+    , aeson                     >=2.0.0.0  && <2.4
     , aeson-pretty              >=0.8.7    && <0.9
     -- cookie 0.4.3 is needed by GHC 7.8 due to time>=1.4 constraint
     , cookie                    >=0.4.3    && <0.6
@@ -94,7 +94,7 @@ library
     , unordered-containers      >=0.2.9.0  && <0.3
     , uuid-types                >=1.0.3    && <1.1
     , vector                    >=0.12.0.1 && <0.14
-    , QuickCheck                >=2.10.1   && <2.17
+    , QuickCheck                >=2.10.1   && <2.19
 
   default-language:    Haskell2010
 
@@ -106,7 +106,7 @@ test-suite spec
   -- From library parat
  -- We need aeson's toEncoding for doctests too
   build-depends:
-      base                      >=4.9       && <4.22
+      base                      >=4.9       && <4.23
     , swagger2
     , aeson
     , base-compat-batteries
@@ -127,7 +127,7 @@ test-suite spec
   build-depends:
       hspec                >=2.5.5   && <2.12
     , HUnit                >=1.6.0.0 && <1.7
-    , quickcheck-instances >=0.3.19  && <0.4
+    , quickcheck-instances >=0.3.19  && <0.5
     , utf8-string          >=1.0.1.1 && <1.1
 
   -- https://github.com/haskell/cabal/issues/3708
@@ -146,7 +146,7 @@ test-suite spec
 
 test-suite doctests
   -- See QuickCheck note in https://github.com/phadej/cabal-doctest#notes
-  build-depends:    base                 >=4.9       && <4.22
+  build-depends:    base                 >=4.9       && <4.23
                   , doctest
                   , Glob
                   , QuickCheck

